### PR TITLE
Fixed table of contents button interactions

### PR DESCRIPTION
**Pull request by m1yag1**
* Refactored the content page object to include new regions that
describe the page. The regions are Navbar, Sidebar, Sidebar header, and
Toolbar.
* Refactored the test to account for the differences in resolutions and
the order of toggling the Sidebar. When the resolution is desktop we
click the toc button on the sidebar to close and the toc button on the
toolbar to open. It's the other way around for mobile and tablet
resolutions.

### CODE REVIEW

File: pytest-selenium/pages/content.py [hunk 1]
```diff
@@ -4,46 +4,72 @@
...
+        return self.ToolBar(self)
 
     @property
-    def table_of_contents(self):
-        return self.TableOfContents(self)
+    def sidebar(self):
+        return self.SideBar(self)
+
+    class NavBar(Region):
+        _root_locator = (By.CSS_SELECTOR, "div[class^='NavBar__TopBar']")
```
Comment (TomWoodward) [inline pytest-selenium/pages/content.py]: we should just add data selectors

File: pytest-selenium/tests/test_toc.py [hunk 2]
```diff
@@ -1,23 +1,62 @@
...
-    #   AND: The table of contents toggle button is clicked
     content = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
 
-    toc = content.table_of_contents
-    content.click_table_of_contents_button()
+    toolbar = content.toolbar
+    sidebar = content.sidebar
+
+    # AND: Window width is 1024 or greater (Desktop)
+    if selenium.get_window_size()["width"] > 1024:
```
Comment (TomWoodward) [inline pytest-selenium/tests/test_toc.py]: magic numbers should be defined somewhere, maybe a top level region with a `isDesktop()` or `isMobile()`?

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: pytest-selenium/pages/content.py (hunk 1 above)
Comment (m1yag1) [reply pytest-selenium/pages/content.py]: Can you show me how to add them? I actually don't think this is that terrible. It works really well and I don't see the prefix portion of these elements changing but I could be wrong.

Comment (m1yag1) [reply pytest-selenium/pages/content.py]: The good thing is that the regions are now more specific. I really like that part. This object representation is a lot closer to how the page is really laid out. This is also much better than XPATH.

Comment (m1yag1) [reply pytest-selenium/pages/content.py]: @TomWoodward ping

Comment (TomWoodward) [reply pytest-selenium/pages/content.py]: the class names there are auto generated based on the component names, so if we ever rename a component all the ui tests will break. i would rather use the more explicit coupling of the data selectors.

Comment (m1yag1) [reply pytest-selenium/pages/content.py]: Done.

**Review by TomWoodward — CHANGES_REQUESTED** [review]
this looks fine, but needs the snaps updated

**Review by TomWoodward — APPROVED** [review]

**Comment by TomWoodward** [comment]
@m1yag1 linter errors
```
ERROR: /home/travis/build/openstax/rex-web/src/app/components/NavBar.tsx[86, 25]: " should be '
ERROR: /home/travis/build/openstax/rex-web/src/app/content/components/Sidebar.tsx[254, 58]: " should be '
ERROR: /home/travis/build/openstax/rex-web/src/app/content/components/Toolbar.tsx[146, 23]: " should be '
```

**Pull request merged**